### PR TITLE
Some more improvements in /layouts

### PR DESCRIPTION
**Pull request by frankmayer**
### Summary of Changes
- some type-safe comparisons
- some removal of redundant code -
- fix of misspelled (case) method call
- replacing of deprecated method call

### Testing Instructions
Should not change behavior.
- Code review

### Documentation Changes Required
None

### CODE REVIEW

File: layouts/joomla/pagination/links.php [hunk 1]
```diff
@@ -65,7 +65,7 @@
 			<?php foreach ($pages['pages'] as $k => $page) : ?>
 
 				<?php $output = JLayoutHelper::render('joomla.pagination.link', $page); ?>
-				<?php if (in_array($k, range($range * $step - ($step + 1), $range * $step))) : ?>
+				<?php if (in_array($k, range($range * $step - ($step + 1), $range * $step), true)) : ?>
```
Comment (andrepereiradasilva) [inline layouts/joomla/pagination/links.php]: Is k always an int?

Comment (frankmayer) [inline layouts/joomla/pagination/links.php]: If it gets its ['pages']-data from libraries/cms/pagination/pagination.php, it is an int. I don't see any other implementation of this...
see here: https://github.com/joomla/joomla-cms/blob/ce7de218c1a4c16168873c2056e70417daa04940/libraries/cms/pagination/pagination.php#L816

**Comment by anibalsanchez** [comment]
I have tested this item :white_check_mark: successfully on c6d59889acb8cdc4e5a4ce7ac2d4e9395b3df68c<br /><br />Code Review OK<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13385">issues.joomla.org/tracker/joomla-cms/13385</a>.</sub>

**Comment by frankmayer** [comment]
Conflict resolved. Pls test and RTC,

File: layouts/joomla/edit/params.php [hunk 2]
```diff
@@ -41,8 +41,8 @@
 		// Ensure any fieldsets we don't want to show are skipped (including repeating formfield fieldsets)
 		if ((isset($fieldSet->repeat) && $fieldSet->repeat == true)
 			|| in_array($name, $ignoreFieldsets)
-			|| (!empty($configFieldsets) && in_array($name, $configFieldsets))
-			|| (!empty($hiddenFieldsets) && in_array($name, $hiddenFieldsets))
+			|| (!empty($configFieldsets) && in_array($name, $configFieldsets, true))
+			|| (!empty($hiddenFieldsets) && in_array($name, $hiddenFieldsets, true))
```
Comment (Quy) [inline layouts/joomla/edit/params.php]: Remove unnecessary parentheses.

File: layouts/joomla/edit/params.php [hunk 3]
```diff
@@ -41,8 +41,8 @@
 		// Ensure any fieldsets we don't want to show are skipped (including repeating formfield fieldsets)
 		if ((isset($fieldSet->repeat) && $fieldSet->repeat == true)
```
Comment (Quy) [inline layouts/joomla/edit/params.php]: Change `==` to `===` if it is not done in another PR.

**Comment by Quy** [comment]
Replace of deprecated method call?
https://github.com/joomla/joomla-cms/blob/staging/layouts/joomla/modal/main.php#L153

File: layouts/joomla/edit/params.php (hunk 3 above)
Comment (frankmayer) [inline layouts/joomla/edit/params.php]: Thanks. I probably wasn't sure on that back then. Just rechecked and it seems to be a **real** boolean. Will change that.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 8bec3df1f4e9480fc8b85a8d5d54c755c85eaedd<br /><br />Code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13385">issues.joomla.org/tracker/joomla-cms/13385</a>.</sub>

**Comment by frankmayer** [comment]
Thanks, one more tester to get this RTC?

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on 8bec3df1f4e9480fc8b85a8d5d54c755c85eaedd<br /><br />on code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13385">issues.joomla.org/tracker/joomla-cms/13385</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.
